src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -10066,7 +10066,7 @@ static void VectorizedSmall8(ref T xRef, T y, ref T zRef, ref T dRef, nuint rema
 
         /// <summary>Performs (x * y) + z. It will be rounded as one ternary operation if such an operation is accelerated on the current hardware.</summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static T FusedMultiplyAdd<T>(T x, T y, T addend) where T : INumberBase<T>
+        private static T MultiplyAddEstimate<T>(T x, T y, T addend) where T : INumberBase<T>
```

Comment:
I'd expect this implementation basically boil down to:
```csharp
if (Fma.IsSupported || AdvSimd.IsSupported)
{
    return T.FusedMultiplyAdd(x, y, addend);
}

return (x * y) + addend;
```

That is, use FMA if it's supported (and therefore accelerated), otherwise use the cheaper scalar fallback. 

-- Do we actually need `where T : INumberBase<T>` or can we just use `where T : IFloatingPointIeee754<T>` given the constraint on the public entry point?

Reply:
It's also used by CosineSimilarity, which is only constrained to IRootFunctions

Reply:
Should we be similarly constraining the API here to just `INumberBase<T>` or `IAdditionOperators<T, T, T>, IMultiplyOperators<T, T, T>` then, to either match our own need or to match the `MultiplyAdd` API?

I'd imagine if we need it for something this simple, then it's reasonable to expect other users may similarly need it in lesser constrained scenarios.

Reply:
> Should we be similarly constraining the API here

Which API? The public one? It was approved as being constrained to IFloatingPointIeee754, both for TensorPrimitives and for the addition to the generic math interface.  You're suggesting changing that?

Reply:
I meant for the public one and by extension these internal ones, yes.

I had used `IFloatingPointIeee754` in the proposal under the presumption that most code paths would just use `MultiplyAdd` when dealing with all types and would therefore only need `MultiplyAddEstimate` for the same scenarios where `FusedMultiplyAdd` was available. But `CosineSimilarity` using it, despite being constrained only to `IRootFunctions` (and therefore `INumberBase`) is a good example of where that might be too limiting.

We can always relax it later of course, based on more user input. But we've gone with the "lowest set of constraints" for all the APIs we've exposed so far, so we could just do the same for `MultiplyAddEstimate` and have the publicly exposed constraints match our own internal needs.

Reply:
On which interface would the MultiplyAddEstimate scalar be defined? We'd want our operator's scalar Invoke to call that function, which would mean either constraining that operator's T to wherever MAE is defined, or doing some gymnastics to special-case this.

Reply:
We could define it as low as `INumberBase` (since we need something that includes both `operator *` and `operator +` for the default implementation).

The DIM is the same there as it would be for `IFloatingPointIeee754`. We'd still have the 3 types (`float`, `double`, and `half`) that can accelerate override the behavior to opportunistically use FMA and any user types that can accelerate can do the same.